### PR TITLE
Add Black Formatter

### LINKED ISSUE #152: Add black
Read more about black [here](https://github.com/psf/black).

Sub-Issue of #150

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- [x] Add [black](https://github.com/psf/black)
- [x] Configure pre-commit
- [x] Run on pre-existing files

### Side effects of implementing this change

- n/a

Closes #152

### CODE REVIEW

**Comment by DavisRayM** [comment]
I've run formatted all files locally with black but there seems to be a false positive for "bad continuation line" error on pylint. This is an ongoing discussion? on both [pylint](https://github.com/PyCQA/pylint/issues/289) and [black](https://github.com/psf/black/issues/48).

**Comment by moshthepitt** [comment]
> I've run formatted all files locally with black but there seems to be a false positive for "bad continuation line" error on pylint. This is an ongoing discussion? on both pylint and black.

@DavisRayM lets disable the pylint rule at the affected places in the code.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+max-line-length = 88
```
Comment (DavisRayM) [inline .flake8]: More details on why i did this here https://github.com/psf/black#line-length

**Comment by DavisRayM** [comment]
@moshthepitt Should we let black normalize all string with double quotes. Or should i disable the string normalization ? The PR as is right now normalizes all strings.

**Comment by moshthepitt** [comment]
> @moshthepitt Should we let black normalize all strings with double quotes.

Yes lets let it

**Comment by DavisRayM** [comment]
> > @moshthepitt Should we let black normalize all strings with double quotes.
> 
> Yes lets let it

Nice, this now contains the formatted files. Please have a look and review.

**Pull request merged**